File: pms_backend/src/main/java/com/omar/isdb62/pharmacy_management_backend/model/InvoiceHistory.java
package com.omar.isdb62.pharmacy_management_backend.model;

import jakarta.persistence.*;
import lombok.*;

import java.time.LocalDate;

@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
@Entity
@Table(name = "pms_invoice_history")
public class InvoiceHistory {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    //1
    @Column(name = "invoice_number")
    private String invoiceNumber;

    //2
    @Column(name = "customer_name")
    private String customerName;

    //3
    @Column(name = "contact_number")
    private String contactNumber;

    //4
    @Column(name = "item_name")
    private String itemName;

    //5
    @Column(name = "category")
    private String category;

    //6
    private int quantity;

    //7
    @Column(name = "unit_price")
    private double unitPrice;

    //8
    @Column(name = "sub_total")
    private double subTotal;

    //9
    private double amount;

    //10
    private double discount;

    //11
    @Column(name = "discount_amount")
    private double discountAmount;

    //12
    @Column(name = "net_payable")
    private double netPayable;

    //13
    @Column(name = "date")
    private LocalDate date;  // Add this field

}

File: pms_flutter_app/lib/model/invoice_history.dart
// model/invoice_history.dart
class InvoiceHistory {
  final String? invoiceNumber;
  final String customerName;
  final String contactNumber;
  final String itemName;
  final String category;
  final int quantity;
  final double unitPrice;
  final double subTotal;
  final double amount;
  final double discount;
  final double discountAmount;
  final double netPayable;

  InvoiceHistory({
    this.invoiceNumber,
    required this.customerName,
    required this.contactNumber,
    required this.itemName,
    required this.category,
    required this.quantity,
    required this.unitPrice,
    required this.subTotal,
    required this.amount,
    required this.discount,
    required this.discountAmount,
    required this.netPayable,
  });

  Map<String, dynamic> toJson() => {
    "customerName": customerName,
    "contactNumber": contactNumber,
    "itemName": itemName,
    "category": category,
    "quantity": quantity,
    "unitPrice": unitPrice,
    "subTotal": subTotal,
    "amount": amount,
    "discount": discount,
    "discountAmount": discountAmount,
    "netPayable": netPayable,
  };

  factory InvoiceHistory.fromJson(Map<String, dynamic> json) {
    return InvoiceHistory(
      invoiceNumber: json['invoiceNumber'],
      customerName: json['customerName'],
      contactNumber: json['contactNumber'],
      itemName: json['itemName'],
      category: json['category'],
      quantity: json['quantity'],
      unitPrice: json['unitPrice'],
      subTotal: json['subTotal'],
      amount: json['amount'],
      discount: json['discount'],
      discountAmount: json['discountAmount'],
      netPayable: json['netPayable'],
    );
  }
}
